### PR TITLE
python311Packages.opensfm: unbreak the builds

### PR DESCRIPTION
**Pull request by SomeoneSerge**
## Description of changes

Cf. [an example of the cuda issue](https://hercules-ci.com/accounts/github/SomeoneSerge/derivations/%2Fnix%2Fstore%2F9wjrwsmmb1d78z6m56yd4q1jw66124sb-python3.11-OpenSfM-unstable-2022-03-10.drv/log?via-job=863499dc-27d3-4c7d-b671-b4fedd8258a6) and the [sphinx issue](https://hydra.nixos.org/build/243883830/nixlog/1). Cf. also https://github.com/mapillary/OpenSfM/issues/1027 and https://github.com/mapillary/OpenSfM/issues/1029. Still following refs/main/HEAD for now, but might switch to a release depending on how https://github.com/mapillary/OpenSfM/issues/1028 is resolved

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/opencv/4.x.nix [hunk 1]
```diff
@@ -472,7 +472,12 @@ effectiveStdenv.mkDerivation {
   postInstall = ''
     sed -i "s|{exec_prefix}/$out|{exec_prefix}|;s|{prefix}/$out|{prefix}|" \
       "$out/lib/pkgconfig/opencv4.pc"
-    mkdir $cxxdev
+    mkdir "$cxxdev"
+  ''
+  # Temporary fix for https://github.com/NixOS/nixpkgs/issues/276691
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/opencv/4.x.nix:477]:
```suggestion
  # fix deps not progagating from opencv4.cxxdev if cuda is disabled
  # see https://github.com/NixOS/nixpkgs/issues/276691
```
it is always nice to know what the issue is about without needing to visit it

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 274860` at e326909c run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>26 packages marked as broken and skipped:</summary>
  <ul>
    <li>caffe</li>
    <li>caffe.bin</li>
    <li>caffeWithCuda</li>
    <li>caffeWithCuda.bin</li>
    <li>home-assistant-component-tests.dremel_3d_printer</li>
    <li>home-assistant-component-tests.environment_canada</li>
    <li>obs-studio-plugins.obs-hyperion</li>
    <li>opentoonz</li>
    <li>python310Packages.caffe</li>
    <li>python310Packages.caffe.bin</li>
    <li>python310Packages.caffeWithCuda</li>
    <li>python310Packages.caffeWithCuda.bin</li>
    <li>python310Packages.dm-sonnet</li>
    <li>python310Packages.paddleocr</li>
    <li>python310Packages.stytra</li>
    <li>python310Packages.sunpy</li>
    <li>python310Packages.tensorflow-datasets</li>
    <li>python310Packages.unstructured-inference</li>
    <li>python311Packages.caffe</li>
    <li>python311Packages.caffe.bin</li>
    <li>python311Packages.caffeWithCuda</li>
    <li>python311Packages.caffeWithCuda.bin</li>
    <li>python311Packages.dm-sonnet</li>
    <li>python311Packages.stytra</li>
    <li>python311Packages.sunpy</li>
    <li>python311Packages.tensorflow-datasets</li>
  </ul>
</details>
<details>
  <summary>25 packages failed to build:</summary>
  <ul>
    <li>citrix_workspace (citrix_workspace_23_11_0)</li>
    <li>citrix_workspace_23_02_0</li>
    <li>citrix_workspace_23_07_0</li>
    <li>citrix_workspace_23_09_0</li>
    <li>ib-controller</li>
    <li>mathematica</li>
    <li>mathematica-cuda</li>
    <li>mathematica-webdoc</li>
    <li>mathematica-webdoc-cuda</li>
    <li><a href="https://gist.github.com/natsukium/1c3f480d1fd2b60724be896e88dea0a9">obs-studio-plugins.obs-replay-source</a></li>
    <li>python310Packages.bsuite</li>
    <li><a href="https://gist.github.com/natsukium/e8228a1afaf45c49d5dd8b9b87eacc6b">python310Packages.fairseq</a></li>
    <li>python310Packages.rlax</li>
    <li><a href="https://gist.github.com/natsukium/7b5c24c6076f2f6dc18d4c002ae8b7f6">python310Packages.sfepy</a></li>
    <li>python310Packages.shap</li>
    <li><a href="https://gist.github.com/natsukium/2f43b06f30ba8c55f3617e0cc43cb516">python310Packages.wagtail-localize</a></li>
    <li>python311Packages.bsuite</li>
    <li><a href="https://gist.github.com/natsukium/e7adcc8d235f2c8cc29c550d19a65891">python311Packages.fairseq</a></li>
    <li>python311Packages.rlax</li>
    <li><a href="https://gist.github.com/natsukium/8471c5930c91a09d3ccca61a249fd880">python311Packages.sfepy</a></li>
    <li>python311Packages.shap</li>
    <li><a href="https://gist.github.com/natsukium/5f5b9dfd5d3f9af89593528b4aba9c09">python311Packages.wagtail-localize</a></li>
    <li><a href="https://gist.github.com/natsukium/e28270a47584c6346b09276099e43d44">sitespeed-io</a></li>
    <li>wolfram-engine</li>
    <li>wolfram-notebook</li>
  </ul>
</details>
<details>
  <summary>330 packages built successfully:</summary>
  <ul>
    <li>MIDIVisualizer</li>
    <li>aitrack</li>
    <li>audiobookshelf</li>
    <li>auto-multiple-choice</li>
    <li>cctag</li>
    <li>cctag.dev</li>
    <li>cctag.lib</li>
    <li>corrscope</li>
    <li>darktable</li>
    <li>deepin.dde-gsettings-schemas</li>
    <li>deepin.deepin-camera</li>
    <li>deepin.deepin-image-viewer</li>
    <li>deepin.deepin-ocr-plugin-manager</li>
    <li>deepin.deepin-screen-recorder</li>
    <li>deepin.image-editor</li>
    <li>deface</li>
    <li>digikam</li>
    <li>easyocr (python311Packages.easyocr)</li>
    <li>escrotum</li>
    <li>escrotum.man</li>
    <li>facedetect</li>
    <li>faust2jack</li>
    <li>ffcast</li>
    <li>ffmpeg-full (ffmpeg_6-full)</li>
    <li>ffmpeg-full.bin (ffmpeg_6-full.bin)</li>
    <li>ffmpeg-full.data (ffmpeg_6-full.data)</li>
    <li>ffmpeg-full.dev (ffmpeg_6-full.dev)</li>
    <li>ffmpeg-full.doc (ffmpeg_6-full.doc)</li>
    <li>ffmpeg-full.lib (ffmpeg_6-full.lib)</li>
    <li>ffmpeg-full.man (ffmpeg_6-full.man)</li>
    <li>ffmpeg_4-full</li>
    <li>ffmpeg_4-full.bin</li>
    <li>ffmpeg_4-full.data</li>
    <li>ffmpeg_4-full.dev</li>
    <li>ffmpeg_4-full.doc</li>
    <li>ffmpeg_4-full.lib</li>
    <li>ffmpeg_4-full.man</li>
    <li>ffmpeg_5-full</li>
    <li>ffmpeg_5-full.bin</li>
    <li>ffmpeg_5-full.data</li>
    <li>ffmpeg_5-full.dev</li>
    <li>ffmpeg_5-full.doc</li>
    <li>ffmpeg_5-full.lib</li>
    <li>ffmpeg_5-full.man</li>
    <li>ffsubsync</li>
    <li>fgallery</li>
    <li>flowblade</li>
    <li>focus-stack</li>
    <li>frei0r</li>
    <li>frigate</li>
    <li>gimp-with-plugins</li>
    <li>gimpPlugins.gmic</li>
    <li>glaxnimate</li>
    <li>gmic</li>
    <li>gmic-qt</li>
    <li>gmic.dev</li>
    <li>gmic.lib</li>
    <li>gmic.man</li>
    <li>handbrake</li>
    <li>hikounomizu</li>
    <li>hydrus</li>
    <li>hydrus.doc</li>
    <li>imagination</li>
    <li>intensity-normalization (python311Packages.intensity-normalization)</li>
    <li>jellyfin</li>
    <li>jellyfin-ffmpeg</li>
    <li>jellyfin-ffmpeg.bin</li>
    <li>jellyfin-ffmpeg.data</li>
    <li>jellyfin-ffmpeg.dev</li>
    <li>jellyfin-ffmpeg.doc</li>
    <li>jellyfin-ffmpeg.lib</li>
    <li>jellyfin-ffmpeg.man</li>
    <li>krita</li>
    <li>libsForQt5.kdenlive (plasma5Packages.kdenlive)</li>
    <li>libsForQt5.mauikit-imagetools (plasma5Packages.mauikit-imagetools)</li>
    <li>libsForQt5.mlt (plasma5Packages.mlt)</li>
    <li>libsForQt5.mlt.dev (plasma5Packages.mlt.dev)</li>
    <li>libsForQt5.pix (plasma5Packages.pix)</li>
    <li>soundkonverter (libsForQt5.soundkonverter)</li>
    <li>libyafaray</li>
    <li>liquidsoap</li>
    <li>magnetophonDSP.VoiceOfFaust</li>
    <li>manim</li>
    <li>mavproxy</li>
    <li>meerk40t</li>
    <li>meerk40t-camera</li>
    <li>mlt</li>
    <li>mlt.dev</li>
    <li>monado</li>
    <li>monero-gui</li>
    <li>mxnet</li>
    <li>napari (python311Packages.napari)</li>
    <li>ncnn</li>
    <li>nomacs</li>
    <li>obs-studio</li>
    <li>obs-studio-plugins.advanced-scene-switcher</li>
    <li>obs-studio-plugins.droidcam-obs</li>
    <li>obs-studio-plugins.input-overlay</li>
    <li>obs-studio-plugins.looking-glass-obs</li>
    <li>obs-studio-plugins.obs-3d-effect</li>
    <li>obs-studio-plugins.obs-backgroundremoval</li>
    <li>obs-studio-plugins.obs-command-source</li>
    <li>obs-studio-plugins.obs-freeze-filter</li>
    <li>obs-studio-plugins.obs-gradient-source</li>
    <li>obs-studio-plugins.obs-gstreamer</li>
    <li>obs-studio-plugins.obs-livesplit-one</li>
    <li>obs-studio-plugins.obs-move-transition</li>
    <li>obs-studio-plugins.obs-multi-rtmp</li>
    <li>obs-studio-plugins.obs-mute-filter</li>
    <li>obs-studio-plugins.obs-nvfbc</li>
    <li>obs-studio-plugins.obs-pipewire-audio-capture</li>
    <li>obs-studio-plugins.obs-rgb-levels-filter</li>
    <li>obs-studio-plugins.obs-scale-to-sound</li>
    <li>obs-studio-plugins.obs-shaderfilter</li>
    <li>obs-studio-plugins.obs-source-clone</li>
    <li>obs-studio-plugins.obs-source-record</li>
    <li>obs-studio-plugins.obs-source-switcher</li>
    <li>obs-studio-plugins.obs-teleport</li>
    <li>obs-studio-plugins.obs-text-pthread</li>
    <li>obs-studio-plugins.obs-transition-table</li>
    <li>obs-studio-plugins.obs-tuna</li>
    <li>obs-studio-plugins.obs-vaapi</li>
    <li>obs-studio-plugins.obs-vertical-canvas</li>
    <li>obs-studio-plugins.obs-vintage-filter</li>
    <li>obs-studio-plugins.obs-vkcapture</li>
    <li>obs-studio-plugins.obs-websocket</li>
    <li>obs-studio-plugins.waveform</li>
    <li>obs-studio-plugins.wlrobs</li>
    <li>olive-editor</li>
    <li>ome_zarr (python311Packages.ome-zarr)</li>
    <li>opencomposite-helper</li>
    <li>opencv (opencv4 ,opencv4WithoutCuda)</li>
    <li>opencv.cxxdev (opencv4.cxxdev ,opencv4WithoutCuda.cxxdev)</li>
    <li>opencv.package_tests (opencv4.package_tests ,opencv4WithoutCuda.package_tests)</li>
    <li>openmvs</li>
    <li>opentrack</li>
    <li>openvino</li>
    <li>openvino.python</li>
    <li>osm2pgsql</li>
    <li>peek</li>
    <li>photoprism</li>
    <li>printrun</li>
    <li>python310Packages.aigpy</li>
    <li>python310Packages.albumentations</li>
    <li>python310Packages.aplpy</li>
    <li>python310Packages.auditok</li>
    <li>python310Packages.batchgenerators</li>
    <li>python310Packages.boxx</li>
    <li>python310Packages.bpycv</li>
    <li>python310Packages.camelot</li>
    <li>python310Packages.clarifai</li>
    <li>python310Packages.dask-image</li>
    <li>python310Packages.detectron2</li>
    <li>python310Packages.dremel3dpy</li>
    <li>python310Packages.easyocr</li>
    <li>python310Packages.env-canada</li>
    <li>python310Packages.ffcv</li>
    <li>python310Packages.fpdf2</li>
    <li>python310Packages.glueviz</li>
    <li>python310Packages.glymur</li>
    <li>python310Packages.grad-cam</li>
    <li>python310Packages.gradio</li>
    <li>python310Packages.gradio-client</li>
    <li>python310Packages.gymnasium</li>
    <li>python310Packages.handout</li>
    <li>python310Packages.imagecorruptions</li>
    <li>python310Packages.imageio</li>
    <li>python310Packages.imantics</li>
    <li>python310Packages.imutils</li>
    <li>python310Packages.insightface</li>
    <li>python310Packages.intensity-normalization</li>
    <li>python310Packages.invisible-watermark</li>
    <li>python310Packages.k-diffusion</li>
    <li>python310Packages.labelbox</li>
    <li>python310Packages.layoutparser</li>
    <li>python310Packages.lime</li>
    <li>python310Packages.mlt</li>
    <li>python310Packages.mlt.dev</li>
    <li>python310Packages.mmcv</li>
    <li>python310Packages.mmengine</li>
    <li>python310Packages.moderngl-window</li>
    <li>python310Packages.moviepy</li>
    <li>python310Packages.mung</li>
    <li>python310Packages.muscima</li>
    <li>python310Packages.mxnet</li>
    <li>python310Packages.napari</li>
    <li>python310Packages.napari-console</li>
    <li>python310Packages.napari-svg</li>
    <li>python310Packages.ome-zarr</li>
    <li>python310Packages.omrdatasettools</li>
    <li>python310Packages.open-clip-torch</li>
    <li>python310Packages.opencv4</li>
    <li>python310Packages.opencv4.cxxdev</li>
    <li>python310Packages.opencv4.package_tests</li>
    <li>python310Packages.opensfm</li>
    <li>python310Packages.openvino</li>
    <li>python310Packages.pdf2docx</li>
    <li>python310Packages.pillow-heif</li>
    <li>python310Packages.pims</li>
    <li>python310Packages.psd-tools</li>
    <li>python310Packages.pvextractor</li>
    <li>python310Packages.pydub</li>
    <li>python310Packages.pyfakewebcam</li>
    <li>python310Packages.pyglet</li>
    <li>python310Packages.pyrender</li>
    <li>python310Packages.pytmx</li>
    <li>python310Packages.pyvista</li>
    <li>python310Packages.qudida</li>
    <li>python310Packages.remi</li>
    <li>python310Packages.scikit-image</li>
    <li>python310Packages.shazamio</li>
    <li>python310Packages.skrl</li>
    <li>python310Packages.slicedimage</li>
    <li>python310Packages.spectral-cube</li>
    <li>python310Packages.speechbrain</li>
    <li>python310Packages.test-tube</li>
    <li>python310Packages.torchaudio</li>
    <li>python310Packages.videocr</li>
    <li>python310Packages.vidstab</li>
    <li>python310Packages.wagtail</li>
    <li>python310Packages.wagtail-factories</li>
    <li>python310Packages.webdataset</li>
    <li>python310Packages.willow</li>
    <li>python310Packages.zcs</li>
    <li>python311Packages.aigpy</li>
    <li>python311Packages.albumentations</li>
    <li>python311Packages.aplpy</li>
    <li>python311Packages.auditok</li>
    <li>python311Packages.batchgenerators</li>
    <li>python311Packages.camelot</li>
    <li>python311Packages.clarifai</li>
    <li>python311Packages.dask-image</li>
    <li>python311Packages.detectron2</li>
    <li>python311Packages.dremel3dpy</li>
    <li>python311Packages.env-canada</li>
    <li>python311Packages.ffcv</li>
    <li>python311Packages.fpdf2</li>
    <li>python311Packages.glueviz</li>
    <li>python311Packages.glymur</li>
    <li>python311Packages.grad-cam</li>
    <li>python311Packages.gradio</li>
    <li>python311Packages.gradio-client</li>
    <li>python311Packages.gymnasium</li>
    <li>python311Packages.handout</li>
    <li>python311Packages.imagecorruptions</li>
    <li>python311Packages.imageio</li>
    <li>python311Packages.imantics</li>
    <li>python311Packages.imutils</li>
    <li>python311Packages.insightface</li>
    <li>python311Packages.invisible-watermark</li>
    <li>python311Packages.k-diffusion</li>
    <li>python311Packages.labelbox</li>
    <li>python311Packages.layoutparser</li>
    <li>python311Packages.lime</li>
    <li>python311Packages.mlt</li>
    <li>python311Packages.mlt.dev</li>
    <li>python311Packages.mmcv</li>
    <li>python311Packages.mmengine</li>
    <li>python311Packages.moderngl-window</li>
    <li>python311Packages.moviepy</li>
    <li>python311Packages.mung</li>
    <li>python311Packages.muscima</li>
    <li>python311Packages.mxnet</li>
    <li>python311Packages.napari-console</li>
    <li>python311Packages.napari-svg</li>
    <li>python311Packages.omrdatasettools</li>
    <li>python311Packages.open-clip-torch</li>
    <li>python311Packages.opencv4</li>
    <li>python311Packages.opencv4.cxxdev</li>
    <li>python311Packages.opencv4.package_tests</li>
    <li>python311Packages.opensfm</li>
    <li>python311Packages.openvino</li>
    <li>python311Packages.pdf2docx</li>
    <li>python311Packages.pillow-heif</li>
    <li>python311Packages.pims</li>
    <li>python311Packages.psd-tools</li>
    <li>python311Packages.pvextractor</li>
    <li>python311Packages.pydub</li>
    <li>python311Packages.pyfakewebcam</li>
    <li>python311Packages.pyglet</li>
    <li>python311Packages.pyrender</li>
    <li>python311Packages.pytmx</li>
    <li>python311Packages.pyvista</li>
    <li>python311Packages.qudida</li>
    <li>python311Packages.remi</li>
    <li>python311Packages.scikit-image</li>
    <li>python311Packages.shazamio</li>
    <li>python311Packages.skrl</li>
    <li>python311Packages.slicedimage</li>
    <li>python311Packages.spectral-cube</li>
    <li>python311Packages.speechbrain</li>
    <li>python311Packages.test-tube</li>
    <li>python311Packages.torchaudio</li>
    <li>python311Packages.videocr</li>
    <li>python311Packages.vidstab</li>
    <li>python311Packages.wagtail</li>
    <li>python311Packages.wagtail-factories</li>
    <li>python311Packages.webdataset</li>
    <li>python311Packages.willow</li>
    <li>qimgv</li>
    <li>quirc</li>
    <li>rclip</li>
    <li>realesrgan-ncnn-vulkan</li>
    <li>restream</li>
    <li>rocmPackages.mivisionx (rocmPackages.mivisionx-hip ,rocmPackages_5.mivisionx ,rocmPackages_5.mivisionx-hip)</li>
    <li>rocmPackages.mivisionx-cpu (rocmPackages_5.mivisionx-cpu)</li>
    <li>rocmPackages.mivisionx-opencl (rocmPackages_5.mivisionx-opencl)</li>
    <li>rofi-screenshot</li>
    <li>rtabmap</li>
    <li>run-scaled</li>
    <li>saga</li>
    <li>sdrangel</li>
    <li>shaq</li>
    <li>shotcut</li>
    <li>siril</li>
    <li>synfigstudio</li>
    <li>tartube</li>
    <li>tartube-yt-dlp</li>
    <li>tidal-dl</li>
    <li>tone</li>
    <li>tplay</li>
    <li>tts</li>
    <li>ueberzugpp</li>
    <li>video2midi</li>
    <li>vokoscreen</li>
    <li>waifu2x-converter-cpp</li>
    <li>xpano</li>
    <li>xpra</li>
    <li>xpraWithNvenc</li>
    <li>ytfzf</li>
  </ul>
</details>

**Comment by SomeoneSerge** [comment]
The Ofborg failures are present in master: https://github.com/NixOS/nixpkgs/issues/277812

**Review by natsukium — APPROVED** [review]
There seem to be no regressions.
The fix looks fine to me.

**Pull request merged**